pkg/sysregistriesv2/system_registries_v2.go

```diff
@@ -62,11 +63,22 @@ var userRegistriesDir = filepath.FromSlash(".config/containers/registries.conf.d
 // The function errors if the newly created reference is not parsable.
 func (e *Endpoint) rewriteReference(ref reference.Named, prefix string) (reference.Named, error) {
 	refString := ref.String()
-	if !refMatchesPrefix(refString, prefix) {
+	var newNamedRef string
+	// refMatchingPrefix returns the length of the match. Everything that
+	// follows the match gets appended to registries location.
+	prefixLen := refMatchingPrefix(refString, prefix)
+	if prefixLen == -1 {
 		return nil, fmt.Errorf("invalid prefix '%v' for reference '%v'", prefix, refString)
 	}
-
-	newNamedRef := strings.Replace(refString, prefix, e.Location, 1)
+	if e.Location == "" {
+		if e.Insecure {
+			newNamedRef = refString
+		} else {
+			return nil, fmt.Errorf("location is empty as well as insecure field is not set to true")
+		}
```

Comment:
- I don’t see why this should _require_ `Insecure` (e.g. some of earlier discussion today was about the `Blocked` flag instead). “`Location==""` means no rewriting” feels much simpler to explain, to me.
- The input `ref` is already a `reference.Named`, so it can just be returned here without the round-trip through `newNamedRef` if we want to return it unmodified.
- (That could even happen earlier in the function, without even creating `refString`. OTOH enforcing that `refMatchingPrefix` does match is fairly valuable.)

Reply:
RE: Insecure, I could remove it, no worries, but I added it because saw this earlier from @vrothberg :
https://github.com/containers/image/pull/1191#discussion_r609774796 

```
[[registry]]
prefix="*.internal.registry.com
insecure=true
```

I'll simplify the rest based on your comments.

Reply:
@mtrmac returning `ref` directly now. That change will be pushed in the next commit.